packages/core/src/fhir-deduplication/deduplicate-fhir.ts

```diff
@@ -29,350 +30,285 @@ import { createRef } from "./shared";
 export function deduplicateFhir(fhirBundle: Bundle<Resource>): Bundle<Resource> {
   let resourceArrays = extractFhirTypesFromBundle(fhirBundle);
 
-  // TODO: Add unit tests for the ID replacements
-  const danglingLinks: string[] = [];
-
-  // DocumentReference cleaning
-  resourceArrays.documentReferences = processDocumentReferences(resourceArrays.documentReferences);
-
-  // Practitioner deduplication
-  const practitionersResult = deduplicatePractitioners(resourceArrays.practitioners);
-  resourceArrays = replaceResourceReferences(resourceArrays, practitionersResult.refReplacementMap);
-  resourceArrays.practitioners = practitionersResult.combinedResources;
-  danglingLinks.push(...practitionersResult.danglingReferences);
-
-  // Conditions deduplication
-  const conditionsResult = deduplicateConditions(resourceArrays.conditions);
-  resourceArrays = replaceResourceReferences(resourceArrays, conditionsResult.refReplacementMap);
-  resourceArrays.conditions = conditionsResult.combinedResources;
-  danglingLinks.push(...conditionsResult.danglingReferences);
-
-  // Allergies deduplication
-  const allergiesResult = deduplicateAllergyIntolerances(resourceArrays.allergies);
-  resourceArrays = replaceResourceReferences(resourceArrays, allergiesResult.refReplacementMap);
-  resourceArrays.allergies = allergiesResult.combinedResources;
-  danglingLinks.push(...allergiesResult.danglingReferences);
-
   // Medication deduplication
   const medicationsResult = deduplicateMedications(resourceArrays.medications);
-  resourceArrays = replaceResourceReferences(resourceArrays, medicationsResult.refReplacementMap);
+  resourceArrays = replaceResourceReferences(resourceArrays, medicationsResult.refReplacementMap, [
+    "medicationAdministrations",
+    "medicationStatements",
+    "medicationRequests",
+  ]);
```

Comment:
do medications replacement before medAdmin etc. since med is in the key

Reply:
nit: I think there should be a comment in code for this. Currently, I feel like we're the only two ppl who are aware of this intricacy of the flow